### ISSUE TITLE
[Feature request] Ability to scan credit cards with phone camera

### ISSUE THREAD

**Issue by StefanWallin** [Adyen/adyen-react-native#541]
![Credit Card Scanning demo with react-native-cardscan package](https://raw.githubusercontent.com/getbouncer/react-native-cardscan/HEAD/docs/images/demo.gif)

**Is your feature request related to a problem? Please describe.**
Not all banks support Apple pay or Google Pay, sometimes you want to pay with credit card. Entering the credit cards manually is a pain for the user which can result in a higher cart abandonment percentage.

**Describe the solution you'd like**
To be able to bring our users a delightful experience it would be nice if adyen react native supported scanning a payment card with a card scanner to automatically enter the digits for the user. Something like what https://www.npmjs.com/package/react-native-cardscan can do for you.

**Describe alternatives you've considered**
I've considered implementing https://www.npmjs.com/package/react-native-cardscan myself but find no way to pass the card details into adyen dropin. Preferrably this would be part of the Drop-in dialog.

**Additional context**
When testing and designing this feature one need to take into consideration the "new" quick read designs as well as traditional embossed and raised numbers.

**VISA**
<img width="796" alt="Screenshot 2024-10-09 at 10 25 21" src="https://github.com/user-attachments/assets/de8fa102-6c98-4365-b05f-0454296cdd24">
 
[Visa Brand Mark and Card Design Features](https://bd.visa.com/dam/VCOM/download/merchants/New_VBM_Acq_Merchant_62714_v5.pdf)

[Discover Card Design Guidelines](https://www.discoverglobalnetwork.com/content/dam/discover/en_us/dgn/docs/card-identification-features.pdf)

**Comment by descorp** [comment]
Hey @StefanWallin 

Card reading capability is something we are actively working on ATM.
Thank you for sharing your research! 💚 

I'll pass the links and will keep you posted.

**Comment by jonathanribas** [comment]
Hi @descorp, are you still working on it? 
Any chance it's get implemented on https://github.com/Adyen/adyen-web too?
Thank you!

**Comment by descorp** [comment]
Hey @jonathanribas 

We are piloting an in-house-built card scanning tool and plan to introduce it publicly with the new mobile v6 release in early 2026.

---

> Any chance it's get implemented on https://github.com/Adyen/adyen-web too?

Card scanning is already supported out of the box for mobile browsers. This feature is less relevant for desktop users, so we are not planning to work on it for Web SDK.

**Comment by jonathanribas** [comment]
Hi @descorp, thanks for your reply!
Do you mean https://github.com/Adyen/adyen-web with `new mobile v6`?

**Comment by descorp** [comment]
> Do you mean https://github.com/Adyen/adyen-web with new mobile v6?

No, by "mobile v6" I mean a release of the [adyen-iOS](https://github.com/Adyen/adyen-ios) SDK v6.0.0 and [adyen-android](https://github.com/Adyen/adyen-android) SDK v6.0.0.
Existing or future versions of the adyen-web SDK will not be affected.

**Comment by SpenderJules** [comment]
Hello @descorp , thanks a lot for providing these informations !
Would you be able to provide any expected timeline regarding the delivery of the v6.0.0 of the mobile SDK ?

**Comment by camil-adyen** [comment]
Hi @SpenderJules, v6.0.0 of the mobile SDKs is planned for Q2 '26. However, we are planning a pilot for the card scanning feature before that date. Are you already processing LIVE traffic with the React Native SDK? If so, please ask your Adyen account manager to reach out to the Checkout team if you are interested in joining the pilot and testing this feature together with us. We can assess the pilot criteria and see if it's a match.
